packages/ionic/test/number-control.spec.ts

```diff
@@ -0,0 +1,221 @@
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import { NumberControlRenderer, numberControlTester } from '../src';
+import { By } from '@angular/platform-browser';
+import { MockNgRedux } from '@angular-redux/store/testing';
+import {
+    additionalProps,
+    additionalTestData,
+    canBeDisabled,
+    defaultTestData,
+    ErrorTestExpectation,
+    mustHaveId,
+    numberAdditionalPropsTest,
+    numberErrorTest,
+    numberInputEventTest,
+    renderFloat,
+    renderInteger,
+    setupMockStore,
+    showErrors,
+    updateFloatState,
+    updateWithNull,
+    updateWithSiblingValue,
+    updateWithUndefined
+} from '@jsonforms/angular-test';
+import { NgRedux } from '@angular-redux/store';
+import { IonicModule, Label, Platform, TextInput } from 'ionic-angular';
+import { PlatformMock } from '../test-config/platform-mock';
+import { async, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { Subject } from 'rxjs';
+
+describe('Number field tester', () => {
+    const uischema = {
+        type: 'Control',
+        scope: '#/properties/foo'
+    };
+
+    it('should succeed with floats', () => {
+        expect(
+            numberControlTester(
+                uischema,
+                {
+                    type: 'object',
+                    properties: {
+                        foo: { type: 'number' }
+                    }
+                }
+            )
+        ).toBe(2);
+    });
+    it('should succeed with integers', () => {
+        expect(
+            numberControlTester(
+                uischema,
+                {
+                    type: 'object',
+                    properties: {
+                        foo: { type: 'integer' }
+                    }
+                }
+            )
+        ).toBe(2);
+    });
+});
+
+describe(
+    'Ionic number control', () => {
+        let fixture;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [NumberControlRenderer],
+                imports: [IonicModule.forRoot(NumberControlRenderer)],
+                providers: [
+                    { provide: Platform, useClass: PlatformMock },
+                    { provide: NgRedux, useFactory: MockNgRedux.getInstance }
+                ]
+            }).compileComponents();
+
+            MockNgRedux.reset();
+            fixture = TestBed.createComponent(NumberControlRenderer);
+        });
+
+        it('should render floats', () => {
+            renderFloat(fixture, defaultTestData, () => {
+                const textInput: DebugElement = fixture.debugElement.query(By.directive(TextInput));
+                const textInputInstance = textInput.componentInstance;
```

Comment:
`const textInput: DebugElement = fixture.debugElement.query(By.directive(TextInput));
 const textInputInstance = textInput.componentInstance;
`
We could extract this